File: client-clojure/src/sirqul_io_t_platform/specs/weather_response.clj
(ns sirqul-io-t-platform.specs.weather-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.weather-forecast-response :refer :all]
            )
  (:import (java.io File)))


(def weather-response-data
  {
   (ds/opt :id) int?
   (ds/opt :cloudcover) int?
   (ds/opt :humidity) int?
   (ds/opt :observationTime) int?
   (ds/opt :precipMM) float?
   (ds/opt :pressure) int?
   (ds/opt :temp_C) int?
   (ds/opt :temp_F) int?
   (ds/opt :visibility) int?
   (ds/opt :weatherCode) int?
   (ds/opt :weatherDesc) string?
   (ds/opt :weatherIconUrl) string?
   (ds/opt :winddir16Point) string?
   (ds/opt :winddirDegree) int?
   (ds/opt :windspeedKmph) int?
   (ds/opt :windspeedMiles) int?
   (ds/opt :forecasts) (s/coll-of weather-forecast-response-spec)
   })

(def weather-response-spec
  (ds/spec
    {:name ::weather-response
     :spec weather-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/WeatherResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import WeatherResponse._

case class WeatherResponse (
  id: Option[Long],
cloudcover: Option[Integer],
humidity: Option[Integer],
observationTime: Option[Long],
precipMM: Option[Double],
pressure: Option[Integer],
tempC: Option[Integer],
tempF: Option[Integer],
visibility: Option[Integer],
weatherCode: Option[Integer],
weatherDesc: Option[String],
weatherIconUrl: Option[String],
winddir16Point: Option[String],
winddirDegree: Option[Integer],
windspeedKmph: Option[Integer],
windspeedMiles: Option[Integer],
forecasts: Option[List[WeatherForecastResponse]])

object WeatherResponse {
  import DateTimeCodecs._

  implicit val WeatherResponseCodecJson: CodecJson[WeatherResponse] = CodecJson.derive[WeatherResponse]
  implicit val WeatherResponseDecoder: EntityDecoder[WeatherResponse] = jsonOf[WeatherResponse]
  implicit val WeatherResponseEncoder: EntityEncoder[WeatherResponse] = jsonEncoderOf[WeatherResponse]
}
